driver/format/RowBinaryWithNamesAndTypes.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "driver/platform/platform.h"
+#include "driver/result_set.h"
+
+class RowBinaryWithNamesAndTypesResultSet
```

Comment:
Could you please add a comment explaining what is the solve purpose of this `ResultSet` subclass and how it is different from any other `ResultSet` ?